src/Plugin/Commerce/PaymentGateway/PaypalPaymentGatewayTrait.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Drupal\commerce_paypal\Plugin\Commerce\PaymentGateway;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Provides common methods to be used by PayPal payment gateways.
+ */
+trait PaypalPaymentGatewayTrait {
+
+  /**
+   * Loads the payment for a given remote id.
+   *
+   * @param string $remote_id
+   *   The remote id property for a payment.
+   *
+   * @return \Drupal\commerce_payment\Entity\PaymentInterface
+   *   Payment object.
+   */
+  public function loadPaymentByRemoteId($remote_id) {
```

Comment:
Is there an issue to port this to the payment storage? If so, let's make a `@todo` referencing it. If not, make the issue and make the `@todo` :)